components/ai_chat/resources/page/state/data-context-provider.tsx

```diff
@@ -163,17 +163,17 @@ function DataContextProvider (props: DataContextProviderProps) {
     getPageHandlerInstance().pageHandler.getShouldSendPageContents().then(({ shouldSend }) => setShouldSendPageContents(shouldSend))
   }
 
-  const shouldShowLongPageWarning = React.useMemo(() => {
+  React.useEffect(() => {
+    if (shouldShowLongPageWarning) return
+
     if (
-      !hasDismissedLongPageWarning &&
+      !isGenerating &&
       conversationHistory.length >= 1 &&
-      siteInfo?.isContentTruncated
+      siteInfo?.contentUsedPercentage < 100 && siteInfo?.contentUsedPercentage > 0
     ) {
-      return true
+      setShouldShowLongPageWarning(true)
     }
-
-    return false
-  }, [conversationHistory, hasDismissedLongPageWarning, siteInfo?.isContentTruncated])
+  }, [conversationHistory, siteInfo?.contentUsedPercentage, isGenerating, shouldShowLongPageWarning])
```

Comment:
Sorry for just noticing this but, if this message isn't dismissable then why is it a `useState` and a `useEffect` rather than something computed (memo-ized or not)? e.g.

```ts
const shouldShowLongPageWarning = React.useMemo(() => 
  !isGenerating &&
    conversationHistory.length >= 1 &&
    siteInfo?.contentUsedPercentage < 100
, [isGenerating, conversationHistory.length, siteInfo?.contentUsedPercentage])
```

Reply:
(also don't need to check if above `0` anymore)

Reply:
the reason I wanted to keep a state and effect was to have it flipped only once, as opposed to useMemo, which will recompute based on those dependencies and rerender. the side effect of this was that any subsequent messages would make the long page info fade in and out, i.e., render again.

Reply:
If I change model to one that uses more content, shouldn't this be re-evaluated and the message disappear?

Reply:
why should it disappear? shouldn't we preserve that information along with the user message that was sent to trigger it? it may make sense when we implement conversation dividers, as shown below, since it will be easier to grasp how much content was scanned with respect to the model choice.

<img width="401" alt="image" src="https://github.com/brave/brave-core/assets/8665427/4beb77f9-1669-4f06-9e97-b9d2652ea926">

however, if this info needs to always be up to date, then its position should be changed, and it shouldn't anchor a conversation message. anchoring it to a conversation message implies it's sent and locked in place. maybe it could dynamically update or float near the most current interaction, reflecting real-time changes without being tied to any specific message. this way, it remains relevant and accurately represents the current state of the conversation.

Reply:
well yeah, when you change the model mid-conversation, it now could send the whole article text if the model has a higher limit. So we don't need to show the warning anymroe. Yes it would be cleaner with a new conversation, but it seems appropriate to remove the warning for an existing conversation at that point too since future messages in the conversation will not be limited, and presumably the user is likely to have seen the warning that applies to the first conversation entry, as the user subsequently changed model.

Reply:
Would we even be able to keep the current conversation active if the user goes from free to premium? Like, go through all the flow, load the premium credentials and come back to this open conversation without reloading it?

Reply:
yes, i don't see why wouldn't it be possible to keep the conversation active when user switches from basic to premium?